### PR TITLE
[Form] Fixed violation mapping if multiple forms are using the same (or part of the same) property path

### PR DESCRIPTION
**Pull request by alekitto**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | #5656 |
| License | MIT |
| Doc PR |  |

### CODE REVIEW

File: src/Symfony/Component/Form/Extension/Validator/ViolationMapper/ViolationMapper.php [hunk 1]
```diff
@@ -196,33 +187,27 @@ private function matchChild(FormInterface $form, PropertyPathIteratorInterface $
...
-                        $foundChild = $child;
-                        $foundAtIndex = $it->key();
-                    }
+            /** @var FormInterface $child */
+            foreach ($children as $key => $child) {
+                $childPath = (string) $child->getPropertyPath();
+                if ($childPath === $chunk) {
+                    $target = $child;
+                    $foundAtIndex = $it->key();
+                } elseif (strpos($childPath, $chunk) === 0) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Extension/Validator/ViolationMapper/ViolationMapper.php]: we usually prefer yoda style (swapping left and right hands of comparisons)

File: src/Symfony/Component/Form/Extension/Validator/ViolationMapper/ViolationMapper.php [hunk 2]
```diff
@@ -196,33 +187,27 @@ private function matchChild(FormInterface $form, PropertyPathIteratorInterface $
...
-            // If we reached the end of the path or if there are no
-            // more matching mapping rules, return the found child
-            if (null !== $foundChild && (!$it->valid() || count($rules) === 0)) {
-                // Reset index in case we tried to find mapping
-                // rules further down the path
-                $it->seek($foundAtIndex);
-
-                return $foundChild;
-            }
+        if ($foundAtIndex !== null) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Extension/Validator/ViolationMapper/ViolationMapper.php]: yoda style

**Comment by alekitto** [comment]
@nicolas-grekas: ok, comparisons edited as you suggested

**Comment by fabpot** [comment]
Thank you @alekitto.

**Pull request closed without merging**